web/main/legal_document_sources.py

```diff
@@ -576,38 +580,124 @@ def pull(legal_doc_source, id):
             )
             resp.raise_for_status()
             cluster = resp.json()
-            resp = requests.get(
-                f"{settings.COURTLISTENER_BASE_URL}/api/rest/v3/opinions/{id}/",
-                headers={"Authorization": f"Token {settings.COURTLISTENER_API_KEY}"},
-            )
-            resp.raise_for_status()
+            cluster["html_info"] = {"source": "court listener"}
+            cluster["sub_opinions"].sort(key=lambda url: int(url.split("/")[-2]))
+
+            sub_opinion_jsons = []
+            for opinion in cluster["sub_opinions"]:
+                sub_opinion_jsons.append(CourtListener.get_opinion_body(opinion))
+
+            text_source = ""
+            for content_type in ("xml_harvard", "html", "plain_text"):
```

Comment:
Looking at https://www.courtlistener.com/help/api/rest/v3/case-law/#opinion-endpoint , I'm realizing that "html" and "plain_text" are the least preferred fields, and html_with_citations is the most preferred field. What if we use this preference:

- First prefer xml_harvard, because we already know how to render it nicely on h2o.
- Then fields in the preferred order at that link starting with html_with_citations.

It looks to me like the citations markup is harmless and we could just pass it through.

Reply:
Cool, I just added all text field options in the order CL specifies with the exception of `xml_harvard`.